MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.cpp

```diff
@@ -0,0 +1,229 @@
+    const double krel_min = config.getConfigParameter<double>("krel_min");
+
+    return std::unique_ptr<RelativePermeability>(
+        new NonWettingPhaseVanGenuchten(Sr, Smax, m, krel_min));
+}
+
+/**
+    \param config ConfigTree object which contains the input data
+                  including <type>WettingPhaseBrookCoreyOilGas</type>
+                  and it has a tag of <relative_permeability>
+*/
+std::unique_ptr<RelativePermeability> createWettingPhaseBrookCoreyOilGas(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__type}
+    config.checkConfigParameter("type", "WettingPhaseBrookCoreyOilGas");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__WettingPhaseBrookCoreyOilGas__sr}
+    const double Sr = config.getConfigParameter<double>("sr");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__WettingPhaseBrookCoreyOilGas__smax}
+    const double Smax = config.getConfigParameter<double>("smax");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__WettingPhaseBrookCoreyOilGas__m}
+    const double m = config.getConfigParameter<double>("m");
+    if (m < 1.0)  // m >= 1
+    {
+        OGS_FATAL(
+            "The exponent parameter of WettingPhaseBrookCoreyOilGas\n"
+            "relative permeability model, m, must not be smaller than 1");
+    }
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__WettingPhaseBrookCoreyOilGas__krel_min}
+    const double krel_min = config.getConfigParameter<double>("krel_min");
+
+    return std::unique_ptr<RelativePermeability>(
+        new WettingPhaseBrookCoreyOilGas(Sr, Smax, m, krel_min));
+}
+
+/**
+    \param config ConfigTree object which contains the input data
+                  including <type>NonWettingPhaseBrookCoreyOilGas</type>
+                  and it has a tag of <relative_permeability>
+*/
+std::unique_ptr<RelativePermeability> createNonWettingPhaseBrookCoreyOilGas(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__type}
+    config.checkConfigParameter("type", "NonWettingPhaseBrookCoreyOilGas");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__sr}
+    const double Sr = config.getConfigParameter<double>("sr");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__smax}
+    const double Smax = config.getConfigParameter<double>("smax");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__m}
+    const double m = config.getConfigParameter<double>("m");
+    if (m < 1.0)  // m >= 1
```

Comment:
In other models you check if `m` is  in `[0,1]`. Here you do not do this. So is it allowed to give negative values for `m` in this model?